schematic/utils/schema_utils.py

```diff
@@ -207,15 +214,156 @@ def convert_bool_to_str(provided_bool: bool) -> str:
+            component_names, component_rule = get_component_name_rules(
+                component_names=component_names, component_rule=component_rule
+            )
+
+            # Get rules
+            validation_rules = get_individual_rules(
+                rule=component_rule, validation_rules=validation_rules
+            )
+
+    # Ensure we collected the component names and validation rules like expected
+    try:
+        assert len(component_names) == len(validation_rules)
+    except ValueError:
+        print(
+            f"The number of components names and validation rules does not match "
+            f"for validation rule: {validation_rule_string}."
+        )
+
+    validation_rules_dict = dict(
+        map(lambda i, j: (i, j), component_names, validation_rules)
+    )
+    return validation_rules_dict
+
+
+def parse_single_set_validation_rules(validation_rule_string: str) -> list:
+    # Try to catch an improperly formatted rule
+    if DELIMITERS["component_name_delimiter"] == validation_rule_string[0]:
+        raise ValueError(
+            f"The provided validation rule {validation_rule_string}, looks to be formatted as a component "
+            f"based rule, but is missing the necessary formatting, "
+            f"please refer to the SchemaHub documentation for more details."
+        )
+
+    # Parse rules that are set across *all* components/manifests
+    return validation_rule_string.split(DELIMITERS["rule_delimiter"])
+
+
+def parse_validation_rules(validation_rules: Union[list, dict]) -> Union[list, dict]:
     """Split multiple validation rules based on :: delimiter
     Args:
-        validation_rules, list: list containing a string validation rule
+        validation_rules, Any[List[str], Dict]: List or Dictionary of validation rules,
+            if list, contains a string validation rule; if dictionary, key is the component the
+            rule (value) is applied to
     Returns:
         validation_rules, list: if submitted List
+    Raises:
+        ValueError if Rule is not formatted properly
+    """
+
+    if isinstance(validation_rules, dict):
+        # Rules pulled in as a dict can be used directly
+        return validation_rules
+    elif isinstance(validation_rules, list):
+        # If rules are already parsed from the JSONLD
+        if len(validation_rules) > 1 and isinstance(validation_rules[-1], str):
+            return validation_rules
+        # Parse rules set for a subset of components/manifests
+        elif DELIMITERS["component_rules_delimiter"] in validation_rules[0]:
+            return parse_component_validation_rules(
+                validation_rule_string=validation_rules[0]
+            )
+        # Parse rules that are set across *all* components/manifests
+        else:
+            return parse_single_set_validation_rules(
+                validation_rule_string=validation_rules[0]
+            )
+    else:
+        raise ValueError(
```

Comment:
So this is interesting. Since `validation_rules: Union[list, dict]` can only be a `dict` or `list` technically you don't need this last `else` statement. If other functions are sending things besides dict/list then validation_rules shouldadd those types to the Union, or just have it be `Any`